app/models/audit/ViewPaymentDetails.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.audit
+
+import models.entitlement.LastPaymentFinancialInfo
+import play.api.libs.json.Json
+
+case class ViewPaymentDetailsModel(
+    nino:              String,
+    status:            String,
+    referrer:          String,
+    deviceFingerprint: String,
+    numOfPayments:     Int,
+    payments:          Seq[LastPaymentFinancialInfo]
+)
+
+object ViewPaymentDetailsModel {
+  implicit val formatViewPaymentDetailsModel = Json.format[ViewPaymentDetailsModel]
+  val eventType: String = "ViewPaymentDetails"
```

Comment:
minor [upper camel case](https://docs.scala-lang.org/style/naming-conventions.html#constants-values-and-variables) for constants whenever possible

```suggestion
  val EventType: String = "ViewPaymentDetails"
```